### PR TITLE
Jvenstad/versions in rest api

### PR DESCRIPTION
**Pull request by jonmv**
@smorgrav please review. 
@ldalves FYI.

### CODE REVIEW

File: controller-server/src/test/resources/job/job-type-response.json [hunk 1]
```diff
@@ -8,7 +8,12 @@
           "end":"2018-06-27T10:13:05Z",
           "id":1,
           "targetPlatform": "1.2.3",
-          "targetApplication": "1.0.321-bada55",
+          "targetApplication": {
+            "build": 321,
+            "repository": "repo",
```
Comment (jonmv) [inline controller-server/src/test/resources/job/job-type-response.json]: These are full URLs in reality.

**Comment by jonmv** [comment]
Ping :)

**Review by freva — COMMENTED** [review]
How about also including the `ApplicationVersion.id()`? We include that in the `/application/v4/` api?

**Comment by jonmv** [comment]
Sure, good suggestion.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelper.java [hunk 2]
```diff
@@ -80,10 +80,23 @@ private static void runStatusToSlime(Cursor cursor, RunStatus runStatus, URI bas
 
         cursor.setString("start", runStatus.start().toString());
         cursor.setLong("id", runStatus.id().number());
+
+        cursor.setString("targetPlatform", runStatus.versions().targetPlatform().toString());
+        applicationVersionToSlime(cursor.setObject("targetApplication"), runStatus.versions().targetApplication());
+        runStatus.versions().sourcePlatform().ifPresent(version -> cursor.setString("sourcePlatform", version.toString()));
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelper.java]: You're right, the source versions are not set in the test case :)

[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelper.java]: But ... perhaps we should default to the target versions, when they aren't present. Let's discuss offline.

**Comment by jonmv** [comment]
PTAL.

**Review by smorgrav — APPROVED** [review]

**Pull request merged**